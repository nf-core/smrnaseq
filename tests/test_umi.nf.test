
nextflow_pipeline {

    name "Test Workflow main.nf - test_umi"
    script "main.nf"
    profile "test_umi"
    tag "umi"
    tag "pipeline"

    test("test_umi") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 74 },

                { assert snapshot(
                    path("$outputDir/mirna_quant/bam/mature/SRX8195118_SRR11631014_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/SRX8195118_SRR11631014_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/SRX8195118_SRR11631014_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/SRX8195117_SRR11631013_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/SRX8195117_SRR11631013_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/SRX8195117_SRR11631013_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195117_SRR11631013_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195117_SRR11631013_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195117_SRR11631013_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195118_SRR11631014_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195118_SRR11631014_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/SRX8195118_SRR11631014_mature_hairpin.sorted.flagstat"),
                ).match("mirna_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.txt").exists()
                ).match("mirna_quant_edger_qc") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/mirtop/joined_samples_mirtop.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirna.tsv").exists(),
                ).match("mirna_quant_mirtop") },

                { assert snapshot(
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-report.html").exists(),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-qcstatus.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-mirna-complexity.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/SRX8195117_SRR11631013/mirtrace-stats-rnatype.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-report.html").exists(),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-qcstatus.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-mirna-complexity.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/SRX8195118_SRR11631014/mirtrace-stats-rnatype.tsv")
                ).match("mirtrace") },

                { assert snapshot(
                    path("$outputDir/genome_quant/bam/SRX8195118_SRR11631014_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/SRX8195117_SRR11631013_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/SRX8195117_SRR11631013_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/SRX8195118_SRR11631014_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/SRX8195118_SRR11631014_mature_hairpin_genome.sorted.stats"),
                    path("$outputDir/genome_quant/bam/SRX8195117_SRR11631013_mature_hairpin_genome.sorted.stats")
                ).match("genome_quant_bam") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_report.html").exists()
                ).match("multiqc") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_data/fastqc-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp_filtered_reads_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_complexity_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools-stats-dp.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_After_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_qc_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_length_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_rna_categories_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools_alignment_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_After_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_unique_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_mean_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_contamination_check_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_adapter_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_adapter_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_sources.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_After_filtering.txt")
                ).match("multiqc_multiqc_data") }
            )
        }
    }
}
