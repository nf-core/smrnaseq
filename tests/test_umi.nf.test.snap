{
    "mirtrace": {
        "content": [
            true,
            "mirtrace-stats-contamination_detailed.tsv:md5,c805a20b67de8a9d4bf1d86a5c92a7e8",
            "mirtrace-stats-contamination_basic.tsv:md5,e69bac9cdb8672286714ae046f47ba52",
            "mirtrace-stats-length.tsv:md5,23c93df112644ec817ba44f2d1a288ac",
            "mirtrace-stats-qcstatus.tsv:md5,789cbc8e576cc0ebc1b0d848ef67cc8a",
            "mirtrace-stats-mirna-complexity.tsv:md5,bdf6ab8c20e9a6caa5b00ecd3946fda6",
            "mirtrace-stats-phred.tsv:md5,5facbd5f028d6f8186c7636b8af831f1",
            "mirtrace-stats-rnatype.tsv:md5,c4e4db698aa81f42f49c4d2368a7a735",
            true,
            "mirtrace-stats-contamination_detailed.tsv:md5,ff78979388fffe5e3805a7080e7a0987",
            "mirtrace-stats-contamination_basic.tsv:md5,677d95c9df736edd47db2be36411213c",
            "mirtrace-stats-length.tsv:md5,9e904ff06e6872fea420bd7a743b1fe9",
            "mirtrace-stats-qcstatus.tsv:md5,c457e8b4da79605dbd33b48217b61412",
            "mirtrace-stats-mirna-complexity.tsv:md5,cee0c7d45a47d9a33bbae3df822708db",
            "mirtrace-stats-phred.tsv:md5,d1110059417aabd532a0aaac86db9612",
            "mirtrace-stats-rnatype.tsv:md5,d1dc9a39d78e5c35764e006773bca76d"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T14:33:25.542680002"
    },
    "genome_quant_bam": {
        "content": [
            "SRX8195118_SRR11631014_mature_hairpin_genome.sorted.flagstat:md5,235383f64a943885f5d899f5b8e03eba",
            "SRX8195117_SRR11631013_mature_hairpin_genome.sorted.flagstat:md5,977e88cbe62027285df73e1f7f9cd9bc",
            "SRX8195117_SRR11631013_mature_hairpin_genome.sorted.idxstats:md5,cc0413bf90252c3b3af8926fd64bc873",
            "SRX8195118_SRR11631014_mature_hairpin_genome.sorted.idxstats:md5,a4874de294706a7ead30258944ff2dad",
            "SRX8195118_SRR11631014_mature_hairpin_genome.sorted.stats:md5,6beddbebbd48d4ffc01e059c73a5a330",
            "SRX8195117_SRR11631013_mature_hairpin_genome.sorted.stats:md5,cc5ef7cbd5a29794170ac6cff59dfd76"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T20:24:14.961610878"
    },
    "software_versions": {
        "content": [
            "{BOWTIE_MAP_GENOME={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_HAIRPIN={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_MATURE={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_SEQCLUSTER={bowtie=1.3.0, samtools=1.16.1}, CSVTK_JOIN={csvtk=0.30.0}, DATATABLE_MERGE={r-base=3.6.2}, FASTP={fastp=0.23.4}, FASTP_LENGTH_FILTER={fastp=0.23.4}, FASTQC_RAW={fastqc=0.12.1}, FASTQC_TRIM={fastqc=0.12.1}, FORMAT_HAIRPIN={fastx_toolkit=0.0.14}, FORMAT_MATURE={fastx_toolkit=0.0.14}, INDEX_HAIRPIN={bowtie=1.3.0}, INDEX_MATURE={bowtie=1.3.0}, MIRTOP_COUNTS={mirtop=0.4.28}, MIRTOP_EXPORT={mirtop=0.4.28}, MIRTOP_GFF={mirtop=0.4.28}, MIRTOP_STATS={mirtop=0.4.28}, MIRTRACE_QC={mirtrace=1.0.1}, PARSE_HAIRPIN={seqkit=2.6.1}, PARSE_MATURE={seqkit=2.6.1}, SAMTOOLS_FLAGSTAT={samtools=1.21}, SAMTOOLS_IDXSTATS={samtools=1.21}, SAMTOOLS_INDEX={samtools=1.21}, SAMTOOLS_SORT={samtools=1.21}, SAMTOOLS_STATS={samtools=1.21}, SEQCLUSTER_COLLAPSE={seqcluster=1.2.9}, UMICOLLAPSE_FASTQ={umicollapse=1.0.0-1}, Workflow={nf-core/smrnaseq=v2.3.2dev}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-02T13:51:36.160803304"
    },
    "mirna_quant_bam": {
        "content": [
            "SRX8195118_SRR11631014_mature.sorted.stats:md5,a49f7118b3cba5e304cf018089cefc8e",
            "SRX8195118_SRR11631014_mature.sorted.flagstat:md5,57c6d477394d367ebae59f7267b430a5",
            "SRX8195118_SRR11631014_mature.sorted.idxstats:md5,8b9cf0f1647b938f058b80522df24667",
            "SRX8195117_SRR11631013_mature.sorted.stats:md5,6847689129964daf73e845cd5ee086db",
            "SRX8195117_SRR11631013_mature.sorted.flagstat:md5,171387fb18ba9868e28ca03d24a7daca",
            "SRX8195117_SRR11631013_mature.sorted.idxstats:md5,fb6c4000f82a66654b4f2a40570649b5",
            "SRX8195117_SRR11631013_mature_hairpin.sorted.idxstats:md5,4e7c1c98804febf6210cee5e3941709e",
            "SRX8195117_SRR11631013_mature_hairpin.sorted.flagstat:md5,b86bd14dc687a26ba5a84d1015f4b70a",
            "SRX8195117_SRR11631013_mature_hairpin.sorted.stats:md5,c7d9fb0a872991e46fecdd941085ed9e",
            "SRX8195118_SRR11631014_mature_hairpin.sorted.stats:md5,168c0b770fc77bf9a7f85085aceff75a",
            "SRX8195118_SRR11631014_mature_hairpin.sorted.idxstats:md5,f4485713620f31d97a5006acdf6d8a5d",
            "SRX8195118_SRR11631014_mature_hairpin.sorted.flagstat:md5,e0c44533bc7813d552de4864d997c916"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-19T15:35:36.685710047"
    },
    "mirna_quant_edger_qc": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:47:03.902417016"
    },
    "multiqc_multiqc_data": {
        "content": [
            "fastqc-status-check-heatmap.txt:md5,c1509fdd74b21a718fe099de64514995",
            "fastp_filtered_reads_plot.txt:md5,54caeafa94c6ec8de2e5fda261aee04a",
            true,
            "fastqc-1_sequence_counts_plot.txt:md5,036a1ca02aa27567988d53bcefce1959",
            "mirtrace_complexity_plot.txt:md5,5a860a872f793250b8c4482d031176a8",
            "fastqc-1_per_sequence_gc_content_plot_Percentages.txt:md5,351f949c0abf4fb7587f3f5d9a28d461",
            "multiqc_citations.txt:md5,02ab194a83114a3c2c22c2749cd27717",
            "samtools-stats-dp.txt:md5,74808822577fb62efb39811272e6919e",
            "fastqc_sequence_length_distribution_plot.txt:md5,8c34b57ec084e2da9d62c254c0a517f4",
            "fastp-seq-content-n-plot_Read_1_Before_filtering.txt:md5,dfdb23f41359b8a6b84d6626a0474d02",
            "fastqc-1_sequence_duplication_levels_plot.txt:md5,b5ae95ecd73055798ed70947dda3747c",
            "fastqc-1_per_base_sequence_quality_plot.txt:md5,89adfa92b1cde0ad4e401b430bbc68ce",
            "multiqc_general_stats.txt:md5,ec1bbecb2844d95f6282b893b6709a06",
            "fastqc-1_per_base_n_content_plot.txt:md5,db081d3aa63007e5a78113f0fc26f27d",
            "fastqc_per_base_n_content_plot.txt:md5,5b5b8cee3162d092c0bcddffbd000f34",
            "fastp-seq-quality-plot_Read_1_After_filtering.txt:md5,66a47c7ce00ede2053f8e6eb20ec3417",
            "fastqc_per_sequence_quality_scores_plot.txt:md5,3aa99649540afc898d32d2e49a364487",
            "mirtrace_qc_plot.txt:md5,98a104b1e65164016ae4081b8815f33e",
            "fastqc-1_per_sequence_quality_scores_plot.txt:md5,4108da6fe352558a652ee2b17d609e07",
            "mirtrace_length_plot.txt:md5,440a84ce9bbdb89b736e4e2446382665",
            true,
            "fastqc-1-status-check-heatmap.txt:md5,cb2ea844834808ae4c95c6440269cf2e",
            "fastqc_sequence_counts_plot.txt:md5,78f80dbdcc711e490c779a998f94b69a",
            "mirtrace_rna_categories_plot.txt:md5,57039a101b0062b1849dadf994df3a88",
            "fastp-seq-quality-plot_Read_1_Before_filtering.txt:md5,f36b7cfd3057b26281367397db45033a",
            "samtools_alignment_plot.txt:md5,b5d5a2f86d2b715f310fa8d6a008123d",
            "fastqc_per_base_sequence_quality_plot.txt:md5,e2e187bc0b0c1f0d1abb3b666945c7b3",
            "mirtop_read_count_plot.txt:md5,1a250ab7f8cab415c06fbc16714a1be7",
            "fastp-seq-content-n-plot_Read_1_After_filtering.txt:md5,bbad2035ada86867c4ed579a93b78d64",
            "mirtop_unique_read_count_plot.txt:md5,57463270ca5f6b5fec3f67bdf9dd1970",
            "mirtop_mean_read_count_plot.txt:md5,77e38f9e8daea0f3fd2b8161d36403c5",
            "mirtrace_contamination_check_plot.txt:md5,2e4a51b79b8d062ff195822bfd5a91a6",
            "fastqc_adapter_content_plot.txt:md5,de1d7324ff5146b49fc9a2e6d4633962",
            "fastqc_sequence_duplication_levels_plot.txt:md5,fe7598e49f93bb980a7675a2bb4bd3b5",
            "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,c2f2f9282a50c3eef475664cc969b8ec",
            "fastqc-1_per_sequence_gc_content_plot_Counts.txt:md5,15d8fa32e0c11ef0d3d10fc28370972c",
            "fastqc-1_adapter_content_plot.txt:md5,89cd342fdc6fbba5f67078c9a2f0c684",
            "fastp-seq-content-gc-plot_Read_1_Before_filtering.txt:md5,f832e92fb36db181ed1079be110edb2a",
            "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,488e25de89d18d20f29b86f2580a8df9",
            true,
            "fastp-seq-content-gc-plot_Read_1_After_filtering.txt:md5,d673e3b18c40c5af1edccffba386d678"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-20T16:55:26.694937898"
    },
    "multiqc": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:47:04.136759497"
    },
    "mirna_quant_mirtop": {
        "content": [
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-20T19:12:28.290360163"
    }
}